tls/s2n_tls13_secrets.h

```diff
@@ -31,25 +31,31 @@ typedef enum {
 } s2n_extract_secret_type_t;
 
 struct s2n_tls13_secrets {
-    /* Secrets only used during the handshake */
     uint8_t early_secret[S2N_TLS13_SECRET_MAX_LEN];
+
     uint8_t handshake_secret[S2N_TLS13_SECRET_MAX_LEN];
-    uint8_t master_secret[S2N_TLS13_SECRET_MAX_LEN];
-    s2n_extract_secret_type_t secrets_state;
+    uint8_t client_handshake_secret[S2N_TLS13_SECRET_MAX_LEN];
+    uint8_t server_handshake_secret[S2N_TLS13_SECRET_MAX_LEN];
 
-    /* Secrets used after the handshake */
+    uint8_t master_secret[S2N_TLS13_SECRET_MAX_LEN];
     uint8_t client_app_secret[S2N_TLS13_SECRET_MAX_LEN];
     uint8_t server_app_secret[S2N_TLS13_SECRET_MAX_LEN];
     uint8_t resumption_master_secret[S2N_TLS13_SECRET_MAX_LEN];
+
+    s2n_extract_secret_type_t secrets_state;
 };
 
 S2N_RESULT s2n_tls13_empty_transcripts_init();
-S2N_RESULT s2n_tls13_secrets_finish(struct s2n_connection *conn);
 
 S2N_RESULT s2n_tls13_extract_secret(struct s2n_connection *conn, s2n_extract_secret_type_t secret_type);
 S2N_RESULT s2n_tls13_derive_secret(struct s2n_connection *conn, s2n_extract_secret_type_t secret_type,
         s2n_mode mode, struct s2n_blob *secret);
 
+S2N_RESULT s2n_tls13_secrets_update(struct s2n_connection *conn);
```

Comment:
General question: Are these APIs that s2n-quic can use and they are not available to the public? It doesn't look like we're actually changing how the secrets get generated in this PR. Like, how does this give s2n-quic the secrets earlier?

Reply:
Ah I always forget about the secret callbacks. Okay makes sense to me.